app.js

```diff
@@ -1,20 +1,27 @@
-const express = require('express')
-const app = express()
-const getTopics = require('./controller')
-const endPoints = require('./endpoints.json')
+const express = require("express");
+const app = express();
 
-app.get('/api/topics', getTopics) 
+const endPoints = require("./endpoints.json");
+const { getTopics, getArticleById } = require("./controller");
 
-app.get('/api', (req, res) => {
-    res.status(200).send(endPoints)
-})
+app.get("/api/topics", getTopics);
 
-app.all('*', (req, res) => {
-    res.status(404).send({message: "not found"})
-})
+app.get("/api", (req, res) => {
+  res.status(200).send(endPoints);
+});
+
+app.get("/api/articles/:article_id", getArticleById);
 
 app.use((err, req, res, next) => {
-    res.status(500).send({message: "internal server error"})
+    res.status(400).send({message: "Bad request"})
 })
 
-module.exports = app
+app.all("*", (req, res, next) => {
```

Comment:
nice!

Reply:
let's swap this app.all and the app.use above it so we have all the endpoint logic together, then the error handlers grouped together